src/services/atomic-swap.service.js

```diff
@@ -50,10 +52,70 @@
     throw new Error('Unable to create the proposal on the Atomic Swap Service');
   }
 
+  await addListenedProposal(walletId, id, password);
   return { proposalId: id };
 };
 
+/**
+ * Adds a new proposal to the `listenedProposals` map of a wallet.
+ * @param {string} walletId
+ * @param {string} proposalId
+ * @param {string} password
+ * @return {Promise<void>}
+ */
+const addListenedProposal = async (walletId, proposalId, password) => {
+  // Checking if this wallet map has been initialized already
+  if (!walletListenedProposals.has(walletId)) {
+    walletListenedProposals.set(walletId, new Map());
+    // Will also open the websocket channel
+  }
+
+  const listenedProposals = walletListenedProposals.get(walletId);
+  listenedProposals.set(proposalId, { id: proposalId, password });
```

Comment:
I believe this alert is not relevant for this case, since we're not making queries to a NoSQL database, but only setting a single entry on a memory map.